tests/integration/targets/clickhouse_quota/tasks/initial.yml

```
@@ -0,0 +1,406 @@
+
+- name: Test 7 - Check the role was not created in check mode
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 7 - Verify that the role is not present
+  ansible.builtin.assert:
+    that:
+    - result.result == []
+
+# Test 8 - Basic creation with cluster, keyed_by and limits in check mode
+- name: Test 8 - Create a test quota with cluster, keyed_by, limits in check mode
+  register: result
+  check_mode: true
+  community.clickhouse.clickhouse_quota:
+    state: present
+    name: test_quota
+    cluster: test_cluster
+    keyed_by: user_name
+    limits:
+      - interval: 5 minute
+        randomized_start: true
+        max:
+          execution_time: 10.2
+      - interval: 1 day
+        max:
+          query_selects: 10
+          query_inserts: 10
+    apply_to: CURRENT_USER
+
+- name: Test 8 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is changed
+    - result.executed_statements == ['CREATE QUOTA \'test_quota\' ON CLUSTER \'test_cluster\' KEYED BY user_name FOR RANDOMIZED INTERVAL 5 minute MAX execution_time = 10.2, FOR INTERVAL 1 day MAX query_selects = 10, query_inserts = 10 TO CURRENT_USER']
+
+- name: Test 8 - Check the role was not created in check mode
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 8 - Verify that the role is not present
+  ansible.builtin.assert:
+    that:
+    - result.result == []
+
+# Test 9 - Negative test that you can't specify apply_to_mode == all and a list of users
+- name: Test 9 - Negative test that you can't specify apply_to_mode == all and a list of users
+  register: result
+  check_mode: true
+  ignore_errors: true
+  community.clickhouse.clickhouse_quota:
+    state: present
+    name: test_quota
+    apply_to: CURRENT_USER
+    apply_to_mode: all
+
+- name: Test 9 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is failed
+
+# Test 10 - Actual test with basic limits
+- name: Test 10 - Actual test with basic limits
+  register: result
+  community.clickhouse.clickhouse_quota:
+    name: test_quota
+    keyed_by: user_name
+    limits:
+      - interval: 5 minute
+        randomized_start: true
+        max:
+          queries: 100
+    apply_to: default
+
+- name: Test 10 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is changed
+    - result.executed_statements == ['CREATE QUOTA \'test_quota\' KEYED BY user_name FOR RANDOMIZED INTERVAL 5 minute MAX queries = 100 TO default']
+
+- name: Test 10 - Check the role was actually created
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 10 - Verify that the role is present
+  ansible.builtin.assert:
+    that:
+    - result.result == [['test_quota']]
```

```suggestion
- name: Test 10 - Check the role was actually created
  register: result
  community.clickhouse.clickhouse_client:
    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"

- name: Test 10 - Verify that the role is present
  ansible.builtin.assert:
    that:
    - result.result == [['test_quota']]
```
Can we check the following properties here?
```
    limits:
      - interval: 5 minute
        randomized_start: true
        max:
          queries: 100
```